src/rsa/pss_verify_tests.txt

```diff
@@ -0,0 +1,60 @@
+# Test vectors for RSA-PSS padding for variable public modulus lengths
+#
+# Digest = SHAAlg.
+# Len gives the public modulus length in bits
+#
+
+Digest = SHA256
+Msg = e002377affb04f0fe4598de9d92d31d6c786040d5776976556a2cfc55e54a1dcb3cb1b126bd6a4bed2a184990ccea773fcc79d246553e6c64f686d21ad4152673cafec22aeb40f6a084e8a5b4991f4c64cf8a927effd0fd775e71e8329e41fdd4457b3911173187b4f09a817d79ea2397fc12dfe3d9c9a0290c8ead31b6690a6
+Encoded = 19a94c1484f0157058112e62786f1ec0b492993412d3812f5efafb4650907e7620af012af5204491858c284bd335f600ef81e472c6d23823dce9fda2a1744683755236b4f7cad671a5fd4f3715c478ebbb8dcfc807adbf2400da7dd0ca126fb2ce71437f651d4f27f4ebef3ada807e2c03e838586b4e8cfd142d6808b32d349a1cc1515b50028ac52e01d7ced4ca629833d0759627edcd442052ddc75f92916faee4c6021e5e48b5fc29b24032df3c5d8f96ff2b0f5c5c04242f63e2c3666ba7e2c4914b34fbf3961f5ccb5e1e67a169da3825428e72b4f6bc9615177d4b316108cb61903b1842391220f2f591313c2d14c3f8382bdad3270d628b15edc59cbc
+Len = 2048
+Result = P
+
+Digest = SHA256
+Msg = e002377affb04f0fe4598de9d92d31d6c786040d5776976556a2cfc55e54a1dcb3cb1b126bd6a4bed2a184990ccea773fcc79d246553e6c64f686d21ad4152673cafec22aeb40f6a084e8a5b4991f4c64cf8a927effd0fd775e71e8329e41fdd4457b3911173187b4f09a817d79ea2397fc12dfe3d9c9a0290c8ead31b6690a6
+Encoded = c9cbb20ff49558e10505fea39e85e30f64f7b71a6307e51d58d1b09139e623ca00800224f6024fb84c7f5407f3d4045e4474def670e32d62590861e10612567eb8f51ba7d8b095553c214d75bb9960f33aa1aa173d020558e35d2b6d7385bd04223b707083d3311b37fea10f73fcb88ec5a9f1920fdef4dba6965756cf78a89ba4cea0fc29b834849c05fa6f8ba823e9c47df8b563f56aaa9a20f69dbee63bb56306c467963703e6671e1eb2f33021404f8ee9332b364f33b6468272afe85dd0ff8f94e7eb2a28876a975d51ec87af3e5fba4a50374daa2a0cde9eb8b675d9fab3a8ec5deee1cace4acb6e0df2ae12f6c63cf28c96492f1a742dcf91186cc0bc
+Len = 2049
```

Comment:
It has been a while now since I wrote that code, but I think it's correct in this form (per the spec).

From https://tools.ietf.org/html/rfc3447#section-8.1.1:
> Apply the EMSA-PSS encoding operation (Section
      9.1.1) to the message M to produce an encoded message EM of length
      `\ceil ((modBits - 1)/8)` octets such that the bit length of the
      integer OS2IP (EM) (see Section 4.2) is at most `modBits - 1`, where
      modBits is the length in bits of the RSA modulus n:
     `    EM = EMSA-PSS-ENCODE (M, modBits - 1)`.
      Note that the octet length of EM will be one less than k if
      modBits - 1 is divisible by 8 and equal to k otherwise.

(Here k is the length in bytes of the public modulus).

Using your example values, modBits is 2049, k is 257, so the encoded messages should be \ceil((2049 - 1)/8) = 256 bytes. This is is the exceptional case noted in the last sentence.